mlir/lib/Dialect/Tosa/Transforms/TosaInferShapes.cpp

```diff
@@ -39,30 +40,26 @@ namespace {
 // type-inference related interface.
 // When a non-replaceable use is encountered, the value is wrapped in a
 // cast back to the original type after inference.
-bool isReplaceableUser(Operation *user) {
-  // Handle unregistered dialects.
-  if (!user->getDialect())
-    return false;
-
-  return user->getDialect()->getNamespace() ==
-             TosaDialect::getDialectNamespace() ||
+bool canBeRefined(Operation *user) {
+  Dialect *tosaDialect = user->getContext()->getLoadedDialect<TosaDialect>();
```

Comment:
This still results in a string lookup in map per invocation of this function. Using TypeID of the dialect would avoid that.

Reply:
Fixed